js/views/transactions/Transactions.js

```diff
@@ -0,0 +1,113 @@
+import $ from 'jquery';
+import app from '../../app';
+import { capitalize } from '../../utils/string';
+import loadTemplate from '../../utils/loadTemplate';
+import baseVw from '../baseVw';
+import MiniProfile from '../MiniProfile';
+import Purchases from './Purchases';
+import Sales from './Sales';
+import Cases from './Cases';
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    const opts = {
+      initialTab: 'purchases',
+      ...options,
+    };
+
+    super(opts);
+    this._tab = opts.initialTab;
+    this.tabViewCache = {};
+    this.tabViews = {
+      purchases: Purchases,
+      sales: Sales,
+      cases: Cases,
+    };
+  }
+
+  className() {
+    return 'transactionsPage clrS';
+  }
+
+  events() {
+    return {
+      'click .js-tab': 'onTabClick',
+    };
+  }
+
+  onTabClick(e) {
+    const targ = $(e.target).closest('.js-tab');
+    this.selectTab(targ.attr('data-tab'));
+  }
+
+  selectTab(targ, options = {}) {
+    const opts = {
+      addTabToHistory: true,
+      ...options,
+    };
+
+    if (!this.tabViews[targ]) {
+      throw new Error(`${targ} is not a valid tab.`);
+    }
+
+    let tabView = this.tabViewCache[targ];
+
+    if (!this.currentTabView || this.currentTabView !== tabView) {
+      if (opts.addTabToHistory) {
+        // subRoute is anything after the tab in the route, which is something
+        // we want to maintain, e.g:
+        // transactions/<tab>/<slug>/<blah>
+        // the subRoute is '/<slug>/<blah>'
+        const curRoute = location.hash.startsWith('#ob://') ?
+          location.hash.slice(6) : location.hash.slice(1);
+
+        const subRoute = curRoute
+          .split('/')
+          .slice(2)
+          .join('/');
+
+        // add tab to history
+        app.router.navigate(`transactions/${targ}${subRoute ? `/${subRoute}` : ''}`);
+      }
+
+      this.$('.js-tab').removeClass('clrT active');
+      this.$(`.js-tab[data-tab="${targ}"]`).addClass('clrT active');
+
+      if (this.currentTabView) this.currentTabView.$el.detach();
+
+      if (!tabView) {
+        if (this[`create${capitalize(targ)}TabView`]) {
```

Comment:
Yes and I left it in because I may implement some create functions and since it gracefully handles them being there or not, I don't see too much harm in it being there.